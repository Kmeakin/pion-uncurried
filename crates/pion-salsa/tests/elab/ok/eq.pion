enum Eq(T: Type, x: T, y: T) {
    Refl: Eq(T, x, x)
}

// let r = Refl;

// let true_eq_true: Eq(Bool, true, false) = Refl(Bool, false, ?_);

let false_eq_false: Eq(Bool, false, false) = Refl(Bool, false, ?_);