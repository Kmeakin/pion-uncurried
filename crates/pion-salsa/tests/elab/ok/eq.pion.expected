enum Eq(T: Type, x: T, y: T): Type {
    Refl: Eq(T, x, x),
}

let false_eq_false: Eq(Bool, false, false) = Refl(Bool, false, #error);